### ISSUE TITLE
Search Function Fails to Find Object

### ISSUE THREAD

**Issue by ufuk-ergin-carbon** [dbeaver/dbeaver#36263]
### Description

The search function in the app is not able to find the table despite it is seen. Attaching the screen recording. In this case, I am trying to find `BIGDATA_FUSE__AIRFLOW` table. However, it is not able to find it. After locating it in Snowflake's own Web UI search under `DATAPIPE_PROD_RAW_DB`, I highlight it in the attached screen recording.

### DBeaver Version

Version 24.2.5.202411171748

### Operating System

Windows 11 Enterprise

https://github.com/user-attachments/assets/a67b4401-4573-4f3a-be7f-11fd46fb26c3

**Comment by E1izabeth** [comment]
Thanks for the issue. DBeaver search doesn't work for schemas right now. You can see the object types for search at the right side of the DB Metadata wizard

**Comment by ufuk-ergin-carbon** [comment]
Hi @E1izabeth , I am confused. Here I am not searching for a schema but a table. I just tested the search for other tables under the default schema, it is completely broken. Nothing comes up in the search results.

**Comment by E1izabeth** [comment]
Maybe we didn't understand the case clearly, but from our perspective in the video, you show how to search for a schema name and find nothing.
![image](https://github.com/user-attachments/assets/b9617dcd-0fab-4f07-824a-2e93922a1129)

**Comment by E1izabeth** [comment]
We will try reproducing the issue. Thanks for clarification

**Comment by ufuk-ergin-carbon** [comment]
Right, sorry for the confusion. I am attaching another screen recording. Here, I am searching for FACT_MESSAGES whose full path is: ANALYTICS_PROD_DB.ANALYTICS.FACT_MESSAGES (which is default). As you can see in the recording, nothing comes up.

https://github.com/user-attachments/assets/1d1c8e74-7304-463e-bfb0-ce86d9d8c7fb

Does this help?

**Comment by E1izabeth** [comment]
Yes, now the issue is clear to me. Thanks a lot

**Comment by E1izabeth** [comment]
@ufuk-ergin-carbon, could you tell me how you open the Search dialog?
With the Search button on the main toolbar or with the `Ctrl+H` shortcut?
![image](https://github.com/user-attachments/assets/8f37843b-97eb-444b-a653-0156f15c5b5c)

I'm trying to reproduce the issue with no success for now. I'm trying to notice the difference with your video. For me, the tree is always collapsed when the Search dialog is opened. Maybe the trick to reproduce is somewhere there

**Comment by ufuk-ergin-carbon** [comment]
@E1izabeth , I am opening the dialog using the search button as I didn't know the keyboard shortcut for this. Now, I tried to replicate the issue using both methods, and the issue is still there. I am attaching a screenshot as to how the dialog opens. 
![Screenshot 2024-11-26 204615](https://github.com/user-attachments/assets/5646ab37-0542-428c-b59d-2ff018479f0c)

**Comment by E1izabeth** [comment]
Thank you for quick reply

**Comment by ufuk-ergin-carbon** [comment]
Sure, you're welcome. I am attaching videos using both methods. 

https://github.com/user-attachments/assets/f958aa02-0c02-43d4-805a-3a2626125ee8


https://github.com/user-attachments/assets/e94784c1-9b16-485d-8185-793c68ba1436

**Comment by ufuk-ergin-carbon** [comment]
Does this help?

**Comment by E1izabeth** [comment]
I figured out how to get an expanded tree in the Search dialog—you need to have only one connection in the project. However, it didn't help me reproduce the bug.

Could you provide log files? They are located in the %APPDATA%\DBeaverData\workspace6\\.metadata directory.
Also, you can collect them to a zip archive automatically with `Help->Collect Diagnostic Info`
![image](https://github.com/user-attachments/assets/51c7ca2e-6be2-489e-b495-028b9aa0e386)

**Comment by ufuk-ergin-carbon** [comment]
@E1izabeth , I am attaching the log files.
[dbeaver-diagnostic-info-1732644339389.zip](https://github.com/user-attachments/files/17924057/dbeaver-diagnostic-info-1732644339389.zip)

**Comment by E1izabeth** [comment]
From the logs, I see the following warnings 
`Can't create sql dialect for basic:SQL`
I'm unsure if it might be a reason, but we'll fix it.

Also, from the previous issue #35169, I see that the select query with the table that you can't find using the Search dialog has an error saying that the table is not found. It means that DBeaver didn't get metadata about this table for some reason. Could you try writing a simple select like `select * from FACT_MESSAGES` and check if you have an error for this query in the SQL Editor?

**Comment by E1izabeth** [comment]
P.S. I'm talking about this indication in SQL Editor
![image](https://github.com/user-attachments/assets/658a2426-be94-4cb0-9a5b-1bbbdf8a4e06)

**Comment by ufuk-ergin-carbon** [comment]
The issue that you referenced does not give error for the fact_messages but for SELECT * FROM DATAPIPE_PROD_RAW_DB.MONOLITH__AIRFLOW.SLOTS LIMIT 10;

This is another issue that I reported where the query runs perfectly but there is syntax coloring like the one you see in the screenshot. As you see, there are 10 results in the results tab. But, I ran the following two select statements and there is no error for any. The results for the two came out just fine.

```
select * from FACT_MESSAGES;

SELECT * FROM ANALYTICS_PROD_DB.ANALYTICS.FACT_MESSAGES
```

**Comment by E1izabeth** [comment]
Thank you
